nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -193,8 +220,29 @@ WaypointFollower::followWaypoints()
       }
     } else if (current_goal_status_ == ActionStatus::SUCCEEDED) {
       RCLCPP_INFO(
-        get_logger(), "Succeeded processing waypoint %i, "
-        "moving to next.", goal_index);
+        get_logger(), "Succeeded processing waypoint %i, processing waypoint task execution",
+        goal_index);
+      auto node = shared_from_this();
+      bool is_task_executed = waypoint_task_executor_->processAtWaypoint(
+        goal->poses[goal_index], goal_index);
+      RCLCPP_INFO(
+        get_logger(), "Task execution at waypoint %i %s", goal_index,
+        is_task_executed ? "succeeded" : "failed!");
+      // if task execution was failed and stop_on_failure_ is on , terminate action
+      if ((is_task_executed == false) && (stop_on_failure_ == true)) {
+        RCLCPP_WARN(
+          get_logger(), "Failed to execute task at waypoint %i "
+          " stop on failure is enabled."
+          " Terminating action.", goal_index);
+        result->missed_waypoints = failed_ids_;
```

Comment:
You need to add this waypoint to `failed_ids_`

Reply:
could that be the reason tests failed ?